index.js

```diff
@@ -1,125 +1,201 @@
 'use strict'
 
 const nodePath = require('path')
-
-const commonExtname = require('common-extname')
-const commonPathPrefix = require('common-path-prefix')
-const glob = require('glob')
+const GlobSync = require('glob/sync').GlobSync
+const GLOBSTAR = require('minimatch').GLOBSTAR
+const hasMagic = require('glob').hasMagic
 const identifierfy = require('identifierfy')
-const flatten = require('lodash.flatten')
 
-function getPattern (path, source) {
-  const pattern = source.value.replace(/^glob:/, '').trim()
-  if (!pattern) {
-    throw path.buildCodeFrameError(`Missing glob pattern '${source.value}'`)
+const dirname = nodePath.dirname
+const relative = nodePath.relative
+const resolve = nodePath.resolve
+const fileSeparator = nodePath.sep
+
+function makeRe (set) {
+  let min, max
+  const parts = []
+  for (let i = 0; i < set.length; i++) {
+    const p = set[i]
+    if (typeof p !== 'string') {
+      if (p === GLOBSTAR) {
+        parts.push('(?:(?!(?:/|^)\\.).)*?')
+      } else {
+        parts.push(p._src)
+      }
+      if (min === undefined) {
+        min = i
+      }
+      max = i
+    } else {
+      parts.push(p.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&'))
+    }
   }
-  if (pattern.startsWith('/')) {
-    throw path.buildCodeFrameError(`Glob pattern must be relative, was '${pattern}'`)
+  parts[max] += ')'
+  parts[min] = '(' + parts[min]
+  return '^' + parts.join('/') + '$'
+}
+
+function extLen (set) {
+  const last = set[set.length - 1]
+  if (typeof last === 'string' || last === GLOBSTAR) {
+    return 0
   }
-  return pattern
+  return last._glob.match(/(?:\.[^.*?!+@[\]()]+)*$/)[0].length
 }
 
-function rejectImportDefaultSpecifier (path, specifiers) {
-  for (const specifier of specifiers) {
-    if (specifier.type === 'ImportDefaultSpecifier') {
-      throw path.buildCodeFrameError('Cannot import the default member')
+function generateMembers (gm, cwd) {
+  const found = gm.found
+  const set = gm.minimatch.set[0]
```

Comment:
Hmm. I should be able to get around this by changing  `makeRe`/`makeSubpathExpression` to return an array of regular expressions. I'll need to merge `extLen` into it though. The return would look like: [{regexp: {}, ext: '.js'},...]. Then we just figure out which expression matches each file.